examples/cmd/main.go

```diff
@@ -143,39 +228,99 @@ func (c *DownloaderCmd) ExecuteBody(logger log.Logger, fn bodyFunc) error {
 		if err = database.Migrate(c.DB); err != nil && err != migrate.ErrNoChange {
 			return err
 		}
-
-		downloader, err = github.NewDownloader(client, db)
 	}
 
-	rate0, err := downloader.RateRemaining(context.TODO())
+	downloadersPool, err := c.buildDownloadersPool(logger, db)
 	if err != nil {
 		return err
 	}
-	t0 := time.Now()
 
-	err = fn(client, downloader)
+	rate0, t0, err := c.setUp(downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	err = downloader.SetCurrent(c.Version)
+	err = fn(logger, downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	if c.Cleanup {
-		return downloader.Cleanup(c.Version)
-	}
-
-	elapsed := time.Since(t0)
-
-	rate1, err := downloader.RateRemaining(context.TODO())
+	rate1, t1, err := c.tearDown(downloadersPool)
 	if err != nil {
 		return err
 	}
+
+	elapsed := t1.Sub(t0)
 	rateUsed := rate0 - rate1
 
 	logger.With(log.Fields{"rate-limit-used": rateUsed, "total-elapsed": elapsed}).Infof("All metadata fetched")
```

Comment:
I've just realized that this is wrong now, I'm gonna have a look into this whether we could show the rate limit used for each token.

Reply:
But I guess that it doesn't make much sense given that it is reset every hour. I'd remove this. WDYT?

Reply:
I didn't review code yet but I think `rate-limit-used` and `total-elapsed` are valuable pieces of information at least for debug. So I wouldn't remove them.